digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -525,7 +526,12 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
             }).values.toList
             endLinkOfNewLinks.distinct.size match {
               case 0 => Seq(newLinks.head.copy(discontinuity = discontinuity)) // TODO: Add test case for this.
-              case 1 => newLinks.filterNot(_.equals(endLinkOfNewLinks.head)) :+ endLinkOfNewLinks.head.copy(discontinuity = discontinuity)
+              case 1 => { if (TrackSectionOrder.hasTripleConnectionPoint(newLinks)) {
```

Comment:
rutiruti, indentaatio, suurinpiirtein näin (tähän kommenttiin tän muotoilun asettelu näyttää olevan haastavaa):

```
case 1 => {
              if (TrackSectionOrder.hasTripleConnectionPoint(newLinks)) {
                val lastProjectLink = newLinks.---
                newLinks.filterNot(---
              }
              else
                newLinks.filterNot(---
          }
```